CanvasPlusPlayground/Views/CourseAnnouncementDetailView.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  CourseAnnouncementDetailView.swift
+//  CanvasPlusPlayground
+//
+//  Created by Ethan Fox on 9/13/24.
+//
+
+import SwiftUI
+import UIKit
+
+struct CourseAnnouncementDetailView: View {
```

Comment:
I tried this out in the simulator and it seems like there are lots of AttributeGraph cycles, and that causes a delay when moving into this view. I think the HTML rendering needs to be asynchronous. I would recommend:

```swift
@State private var message = ""

// ...

var body: some View {
       ScrollView {
            VStack {
                Text(announcement.title ?? "NULL_TITLE")
                    .font(.title)
                Text(announcement.createdAt?.formatted() ?? "NULL_DATE")
                    .font(.subheadline)
                AttributedText(.html(withBody: announcement.message ?? "NULL_MESSAGE"))

            }
        }
        .padding()
        .task {
            message = announcement.message?.toHTML() ?? ""
        }
}
```

I tried [this](https://forums.developer.apple.com/forums/thread/677823?answerId=703952022#703952022) too to convert to HTML, but it doesn't keep the formatting. If you wanna keep your method that's fine.

Reply:
I noticed the delay too, and it irked me. I can check out your fix and see if that works better